AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -144,12 +148,25 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
         
         associatedModels = await self.recurseQueryAssociatedModels(modelSchema: self.modelSchema, ids: modelIds)
         
+        if mutationEvent != "MutationEvent" {
+            self.log.debug("[CascadeDelete.1] Start cascade logic for \(mutationEvent) with id \(modelIds)")
+        }
+        
+        let associatedModelNames = associatedModels.compactMap { $1.modelName }
+        
+        if mutationEvent != "MutationEvent" {
+            self.log.debug("[CascadeDelete.2] Query for \(mutationEvent) associated models, found \(associatedModelNames)")
+        }
+
         deletedResult = await withCheckedContinuation { continuation in
             self.storageAdapter.delete(self.modelType,
                                        modelSchema: self.modelSchema,
                                        filter: self.deleteInput.predicate) { result in
                 continuation.resume(returning: result)
             }
+            if mutationEvent != "MutationEvent" {
+                self.log.debug("[CascadeDelete.3] Local cascade delete of \(mutationEvent) successful!")
+            }
```

Comment:
is `storageAdapter.delete` completion closure an escaping closure? if it is, then this code is running before the delete is completed and would be inaccurate. If it's not an escaping closure, then `continuation.resume` is called to resume the suspended task and then the caller resumes while we call this code in parallel to that, which seems error prone. When leveraging async await's `withCheckedContinuation` to convert the completion handler to async should be a simple wrapper that doesn't involve other logic. Also this doesn't check if the `result` is successful or not so it should be based on the `result` being a `.success` case. `deleteResult` (which is the same as `result`, passed back through `continuation.resume(returning: result)`) is passed to `collapseResults` which i think is where this log line can live

Reply:
In `collapseResults`, before returning success, it can log that the local delete was successful

Reply:
i see that this was moved, so will resolve this comment